task-3/tests/groups.test.ts

```diff
@@ -0,0 +1,150 @@
+import dotenv from "dotenv";
+dotenv.config();
+
+import request from "supertest";
+import app from "../app";
+
+import { AuthService } from "../services/authentication";
+import { GroupService } from "../services/group";
+
+const jwtToken = AuthService.generateAccessToken({
+  username: "newUser123",
+  password: "somePassword",
+});
+
+describe("Test Group Path", () => {
+  test("Requesting all groups should return all groups", async () => {
+    const allGroups: any[] = await GroupService.getAllGroups();
+    return request(app)
+      .get("/group/all")
+      .set("Authorization", "Bearer " + jwtToken)
+      .then((response: { statusCode: number; body: any }) => {
+        expect(response.statusCode).toBe(200);
+        expect(Array.isArray(response.body)).toBeTruthy();
+        expect(response.body.length).toEqual(allGroups.length);
+        expect(response.body[0].id).toBe(allGroups[0].id);
+        expect(response.body[0].name).toBe(allGroups[0].name);
+        expect(response.body[0].permissions).toStrictEqual(
+          allGroups[0].permissions
+        );
+      });
+  });
+
+  test("Requesting specific group", async () => {
```

Comment:
Consider using "test.each" and "describe.each" for the repeated endpoint/method combinations.